perf: handleChildrenEventType link old event-type with workflows using create/findMany

ThyMinimalDev opened the pull request:
## What does this PR do?

Links existing (old) event types to newly added workflows in one transaction by finding existing links and bulk-creating only missing ones. This replaces per-pair upserts to cut queries and speed up handleChildrenEventTypes.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.







<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Links existing (old) event types to newly added workflows in one transaction by finding existing links and bulk-creating only missing ones. This replaces per-pair upserts to cut queries and speed up handleChildrenEventTypes.

- **Performance**
  - Find existing workflow–event type pairs once; create missing pairs with createMany.
  - Removes N×M upsert loop and avoids duplicates via pre-filtering.

- **Behavior**
  - Runs only when both currentWorkflowIds and oldEventTypes are present.

<sup>Written for commit f28982d97c1c370c14ff3b54d29a5f603ff48b09. Summary will update automatically on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

ThyMinimalDev committed:
perf: link old event-type with workflows using create/findMany

volnei commented on packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts:
```diff
@@ -326,26 +326,49 @@ export default async function handleChildrenEventTypes({
+        );
+
+        if (newRelationshipsToCreate.length > 0) {
+          await tx.workflowsOnEventTypes.createMany({
```
@cubic-dev-ai are the duplicated ones not being removed by logic?

devin-ai-integration[bot] committed:
test: fix handleChildrenEventTypes test mocks for new workflow linking implementation

Update test to match the new implementation that uses findMany + createMany
instead of upsert for linking workflows to old event types.

- Add mock for workflowsOnEventTypes.findMany to return empty array
- Update assertion from upsert to findMany + createMany

Co-Authored-By: morgan@cal.com <morgan@cal.com>

ThyMinimalDev committed:
Merge branch 'main' into perf-handle-children-event-type-update-old-event-workflows
